// magicbell-js/dist/esm/project-client/index.js
import { BroadcastsService } from './services/broadcasts/index.js';
import { ChannelsService } from './services/channels/index.js';
import { EventsService } from './services/events/index.js';
import { IntegrationsService } from './services/integrations/index.js';
import { UsersService } from './services/users/index.js';
import { WorkflowsService } from './services/workflows/index.js';
export { Environment } from './http/environment.js';
export * from './http/index.js';
export * from './services/broadcasts/index.js';
export * from './services/channels/index.js';
export * from './services/common/index.js';
export * from './services/events/index.js';
export * from './services/integrations/index.js';
export * from './services/users/index.js';
export * from './services/workflows/index.js';
export class Client {
    constructor(config) {
        Object.defineProperty(this, "config", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: config
        });
        Object.defineProperty(this, "broadcasts", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "channels", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "events", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "integrations", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "users", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "workflows", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.broadcasts = new BroadcastsService(this.config);
        this.channels = new ChannelsService(this.config);
        this.events = new EventsService(this.config);
        this.integrations = new IntegrationsService(this.config);
        this.users = new UsersService(this.config);
        this.workflows = new WorkflowsService(this.config);
    }
    set baseUrl(baseUrl) {
        this.broadcasts.baseUrl = baseUrl;
        this.channels.baseUrl = baseUrl;
        this.events.baseUrl = baseUrl;
        this.integrations.baseUrl = baseUrl;
        this.users.baseUrl = baseUrl;
        this.workflows.baseUrl = baseUrl;
    }
    set environment(environment) {
        this.broadcasts.baseUrl = environment;
        this.channels.baseUrl = environment;
        this.events.baseUrl = environment;
        this.integrations.baseUrl = environment;
        this.users.baseUrl = environment;
        this.workflows.baseUrl = environment;
    }
    set timeoutMs(timeoutMs) {
        this.broadcasts.timeoutMs = timeoutMs;
        this.channels.timeoutMs = timeoutMs;
        this.events.timeoutMs = timeoutMs;
        this.integrations.timeoutMs = timeoutMs;
        this.users.timeoutMs = timeoutMs;
        this.workflows.timeoutMs = timeoutMs;
    }
    set token(token) {
        this.broadcasts.token = token;
        this.channels.token = token;
        this.events.token = token;
        this.integrations.token = token;
        this.users.token = token;
        this.workflows.token = token;
    }
}
// c029837e0e474b76bc487506e8799df5e3335891efe4fb02bda7a1441840310c
