// magicbell-js/dist/commonjs/user-client/services/notifications/models/notification.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.notificationRequest = exports.notificationResponse = exports.notification = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.notification = zod_1.z.lazy(() => {
    return zod_1.z.object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        archivedAt: zod_1.z.string().optional().nullable(),
        category: zod_1.z.string().max(100).optional().nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        createdAt: zod_1.z.string(),
        customAttributes: zod_1.z.any().optional().nullable(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        readAt: zod_1.z.string().optional().nullable(),
        seenAt: zod_1.z.string().optional().nullable(),
        sentAt: zod_1.z.string().optional().nullable(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z.string().max(100).optional().nullable(),
        updatedAt: zod_1.z.string(),
        userId: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.notificationResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        action_url: zod_1.z.string().max(2048).optional().nullable(),
        archived_at: zod_1.z.string().optional().nullable(),
        category: zod_1.z.string().max(100).optional().nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        created_at: zod_1.z.string(),
        custom_attributes: zod_1.z.any().optional().nullable(),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        read_at: zod_1.z.string().optional().nullable(),
        seen_at: zod_1.z.string().optional().nullable(),
        sent_at: zod_1.z.string().optional().nullable(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z.string().max(100).optional().nullable(),
        updated_at: zod_1.z.string(),
        user_id: zod_1.z.string(),
    })
        .transform((data) => ({
        actionUrl: data['action_url'],
        archivedAt: data['archived_at'],
        category: data['category'],
        content: data['content'],
        createdAt: data['created_at'],
        customAttributes: data['custom_attributes'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        readAt: data['read_at'],
        seenAt: data['seen_at'],
        sentAt: data['sent_at'],
        title: data['title'],
        topic: data['topic'],
        updatedAt: data['updated_at'],
        userId: data['user_id'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.notificationRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        archivedAt: zod_1.z.string().optional().nullable(),
        category: zod_1.z.string().max(100).optional().nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        createdAt: zod_1.z.string(),
        customAttributes: zod_1.z.any().optional().nullable(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        readAt: zod_1.z.string().optional().nullable(),
        seenAt: zod_1.z.string().optional().nullable(),
        sentAt: zod_1.z.string().optional().nullable(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z.string().max(100).optional().nullable(),
        updatedAt: zod_1.z.string(),
        userId: zod_1.z.string(),
    })
        .transform((data) => ({
        action_url: data['actionUrl'],
        archived_at: data['archivedAt'],
        category: data['category'],
        content: data['content'],
        created_at: data['createdAt'],
        custom_attributes: data['customAttributes'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        read_at: data['readAt'],
        seen_at: data['seenAt'],
        sent_at: data['sentAt'],
        title: data['title'],
        topic: data['topic'],
        updated_at: data['updatedAt'],
        user_id: data['userId'],
    }));
});
