// magicbell-js/dist/commonjs/project-client/services/integrations/models/apns-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.apnsConfigPayloadRequest = exports.apnsConfigPayloadResponse = exports.apnsConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.apnsConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z.string().regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/),
        badge: zod_1.z.string(),
        certificate: zod_1.z
            .string()
            .regex(/^-+?\s?BEGIN PRIVATE KEY-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END PRIVATE KEY+-+\n?$/),
        keyId: zod_1.z.string().min(10).max(10),
        payloadVersion: zod_1.z.string().optional(),
        teamId: zod_1.z.string().min(10).max(10),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z.string().regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/),
        badge: zod_1.z.string(),
        certificate: zod_1.z
            .string()
            .regex(/^-+?\s?BEGIN PRIVATE KEY-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END PRIVATE KEY+-+\n?$/),
        key_id: zod_1.z.string().min(10).max(10),
        payload_version: zod_1.z.string().optional(),
        team_id: zod_1.z.string().min(10).max(10),
    })
        .transform((data) => ({
        appId: data['app_id'],
        badge: data['badge'],
        certificate: data['certificate'],
        keyId: data['key_id'],
        payloadVersion: data['payload_version'],
        teamId: data['team_id'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z.string().regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/),
        badge: zod_1.z.string(),
        certificate: zod_1.z
            .string()
            .regex(/^-+?\s?BEGIN PRIVATE KEY-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END PRIVATE KEY+-+\n?$/),
        keyId: zod_1.z.string().min(10).max(10),
        payloadVersion: zod_1.z.string().optional(),
        teamId: zod_1.z.string().min(10).max(10),
    })
        .transform((data) => ({
        app_id: data['appId'],
        badge: data['badge'],
        certificate: data['certificate'],
        key_id: data['keyId'],
        payload_version: data['payloadVersion'],
        team_id: data['teamId'],
    }));
});
