// magicbell-js/dist/commonjs/project-client/services/integrations/models/apns-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.apnsConfigRequest = exports.apnsConfigResponse = exports.apnsConfig = void 0;
const zod_1 = require("zod");
const apns_config_payload_js_1 = require("./apns-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.apnsConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: apns_config_payload_js_1.apnsConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: apns_config_payload_js_1.apnsConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: apns_config_payload_js_1.apnsConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
