// magicbell-js/dist/commonjs/project-client/services/integrations/models/banner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bannerRequest = exports.bannerResponse = exports.banner = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.banner = zod_1.z.lazy(() => {
    return zod_1.z.object({
        backgroundColor: zod_1.z.string(),
        backgroundOpacity: zod_1.z.number().optional(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.bannerResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        backgroundOpacity: zod_1.z.number().optional(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        backgroundOpacity: data['backgroundOpacity'],
        fontSize: data['fontSize'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.bannerRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        backgroundOpacity: zod_1.z.number().optional(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        backgroundOpacity: data['backgroundOpacity'],
        fontSize: data['fontSize'],
        textColor: data['textColor'],
    }));
});
