// magicbell-js/dist/commonjs/project-client/services/integrations/models/default_.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultRequest = exports.defaultResponse = exports.default_ = void 0;
const zod_1 = require("zod");
const default_hover_js_1 = require("./default-hover.js");
const default_state_js_1 = require("./default-state.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.default_ = zod_1.z.lazy(() => {
    return zod_1.z.object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontFamily: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        hover: default_hover_js_1.defaultHover.optional(),
        margin: zod_1.z.string(),
        state: default_state_js_1.defaultState.optional(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.defaultResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontFamily: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        hover: default_hover_js_1.defaultHoverResponse.optional(),
        margin: zod_1.z.string(),
        state: default_state_js_1.defaultStateResponse.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        borderRadius: data['borderRadius'],
        fontFamily: data['fontFamily'],
        fontSize: data['fontSize'],
        hover: data['hover'],
        margin: data['margin'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.defaultRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontFamily: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        hover: default_hover_js_1.defaultHoverRequest.optional(),
        margin: zod_1.z.string(),
        state: default_state_js_1.defaultStateRequest.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        borderRadius: data['borderRadius'],
        fontFamily: data['fontFamily'],
        fontSize: data['fontSize'],
        hover: data['hover'],
        margin: data['margin'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
