// magicbell-js/dist/commonjs/project-client/services/integrations/models/event-source-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.eventSourceConfigRequest = exports.eventSourceConfigResponse = exports.eventSourceConfig = void 0;
const zod_1 = require("zod");
const event_source_config_payload_js_1 = require("./event-source-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.eventSourceConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: event_source_config_payload_js_1.eventSourceConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.eventSourceConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: event_source_config_payload_js_1.eventSourceConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.eventSourceConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: event_source_config_payload_js_1.eventSourceConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
