// magicbell-js/dist/commonjs/project-client/services/integrations/models/expo-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.expoConfigRequest = exports.expoConfigResponse = exports.expoConfig = void 0;
const zod_1 = require("zod");
const expo_config_payload_js_1 = require("./expo-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.expoConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: expo_config_payload_js_1.expoConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.expoConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: expo_config_payload_js_1.expoConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.expoConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: expo_config_payload_js_1.expoConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
