// magicbell-js/dist/commonjs/project-client/services/integrations/models/fcm-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fcmConfigRequest = exports.fcmConfigResponse = exports.fcmConfig = void 0;
const zod_1 = require("zod");
const fcm_config_payload_js_1 = require("./fcm-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.fcmConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: fcm_config_payload_js_1.fcmConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.fcmConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: fcm_config_payload_js_1.fcmConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.fcmConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: fcm_config_payload_js_1.fcmConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
