// magicbell-js/dist/commonjs/project-client/services/integrations/models/fcm-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.fcmConfigPayloadRequest = exports.fcmConfigPayloadResponse = exports.fcmConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.fcmConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        authProviderX509CertUrl: zod_1.z.string(),
        authUri: zod_1.z.string(),
        clientEmail: zod_1.z.string(),
        clientId: zod_1.z.string(),
        clientX509CertUrl: zod_1.z.string(),
        privateKey: zod_1.z.string().regex(/^-+?\s?BEGIN[A-Z ]+-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END[A-Z ]+-+\n?$/),
        privateKeyId: zod_1.z.string(),
        projectId: zod_1.z.string(),
        tokenUri: zod_1.z.string(),
        type: zod_1.z.string(),
        universeDomain: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.fcmConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        auth_provider_x509_cert_url: zod_1.z.string(),
        auth_uri: zod_1.z.string(),
        client_email: zod_1.z.string(),
        client_id: zod_1.z.string(),
        client_x509_cert_url: zod_1.z.string(),
        private_key: zod_1.z.string().regex(/^-+?\s?BEGIN[A-Z ]+-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END[A-Z ]+-+\n?$/),
        private_key_id: zod_1.z.string(),
        project_id: zod_1.z.string(),
        token_uri: zod_1.z.string(),
        type: zod_1.z.string(),
        universe_domain: zod_1.z.string(),
    })
        .transform((data) => ({
        authProviderX509CertUrl: data['auth_provider_x509_cert_url'],
        authUri: data['auth_uri'],
        clientEmail: data['client_email'],
        clientId: data['client_id'],
        clientX509CertUrl: data['client_x509_cert_url'],
        privateKey: data['private_key'],
        privateKeyId: data['private_key_id'],
        projectId: data['project_id'],
        tokenUri: data['token_uri'],
        type: data['type'],
        universeDomain: data['universe_domain'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.fcmConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        authProviderX509CertUrl: zod_1.z.string(),
        authUri: zod_1.z.string(),
        clientEmail: zod_1.z.string(),
        clientId: zod_1.z.string(),
        clientX509CertUrl: zod_1.z.string(),
        privateKey: zod_1.z.string().regex(/^-+?\s?BEGIN[A-Z ]+-+\n([A-Za-z0-9+\/\r\n]+={0,2})\n-+\s?END[A-Z ]+-+\n?$/),
        privateKeyId: zod_1.z.string(),
        projectId: zod_1.z.string(),
        tokenUri: zod_1.z.string(),
        type: zod_1.z.string(),
        universeDomain: zod_1.z.string(),
    })
        .transform((data) => ({
        auth_provider_x509_cert_url: data['authProviderX509CertUrl'],
        auth_uri: data['authUri'],
        client_email: data['clientEmail'],
        client_id: data['clientId'],
        client_x509_cert_url: data['clientX509CertUrl'],
        private_key: data['privateKey'],
        private_key_id: data['privateKeyId'],
        project_id: data['projectId'],
        token_uri: data['tokenUri'],
        type: data['type'],
        universe_domain: data['universeDomain'],
    }));
});
