// magicbell-js/dist/commonjs/project-client/services/integrations/models/footer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.footerRequest = exports.footerResponse = exports.footer = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.footer = zod_1.z.lazy(() => {
    return zod_1.z.object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.footerResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        borderRadius: data['borderRadius'],
        fontSize: data['fontSize'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.footerRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        borderRadius: zod_1.z.string(),
        fontSize: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        borderRadius: data['borderRadius'],
        fontSize: data['fontSize'],
        textColor: data['textColor'],
    }));
});
