// magicbell-js/dist/commonjs/project-client/services/integrations/models/inbox-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.inboxConfigRequest = exports.inboxConfigResponse = exports.inboxConfig = void 0;
const zod_1 = require("zod");
const inbox_config_payload_js_1 = require("./inbox-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.inboxConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: inbox_config_payload_js_1.inboxConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: inbox_config_payload_js_1.inboxConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: inbox_config_payload_js_1.inboxConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
