// magicbell-js/dist/commonjs/project-client/services/integrations/models/integration-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.integrationConfigRequest = exports.integrationConfigResponse = exports.integrationConfig = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.integrationConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: zod_1.z.any(),
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.integrationConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: zod_1.z.any(),
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.integrationConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: zod_1.z.any(),
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
