// magicbell-js/dist/commonjs/project-client/services/integrations/models/mailgun-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mailgunConfigRequest = exports.mailgunConfigResponse = exports.mailgunConfig = void 0;
const zod_1 = require("zod");
const mailgun_config_payload_js_1 = require("./mailgun-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.mailgunConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: mailgun_config_payload_js_1.mailgunConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.mailgunConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: mailgun_config_payload_js_1.mailgunConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.mailgunConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: mailgun_config_payload_js_1.mailgunConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
