// magicbell-js/dist/commonjs/project-client/services/integrations/models/mailgun-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.mailgunConfigPayloadRequest = exports.mailgunConfigPayloadResponse = exports.mailgunConfigPayload = void 0;
const zod_1 = require("zod");
const mailgun_config_payload_from_js_1 = require("./mailgun-config-payload-from.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.mailgunConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        apiKey: zod_1.z.string().min(1),
        domain: zod_1.z.string().min(1),
        from: mailgun_config_payload_from_js_1.mailgunConfigPayloadFrom.optional(),
        region: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.mailgunConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        api_key: zod_1.z.string().min(1),
        domain: zod_1.z.string().min(1),
        from: mailgun_config_payload_from_js_1.mailgunConfigPayloadFromResponse.optional(),
        region: zod_1.z.string(),
    })
        .transform((data) => ({
        apiKey: data['api_key'],
        domain: data['domain'],
        from: data['from'],
        region: data['region'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.mailgunConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        apiKey: zod_1.z.string().min(1),
        domain: zod_1.z.string().min(1),
        from: mailgun_config_payload_from_js_1.mailgunConfigPayloadFromRequest.optional(),
        region: zod_1.z.string(),
    })
        .transform((data) => ({
        api_key: data['apiKey'],
        domain: data['domain'],
        from: data['from'],
        region: data['region'],
    }));
});
