// magicbell-js/dist/commonjs/project-client/services/integrations/models/ping-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pingConfigRequest = exports.pingConfigResponse = exports.pingConfig = void 0;
const zod_1 = require("zod");
const ping_config_payload_js_1 = require("./ping-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.pingConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: ping_config_payload_js_1.pingConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.pingConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: ping_config_payload_js_1.pingConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.pingConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: ping_config_payload_js_1.pingConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
