// magicbell-js/dist/commonjs/project-client/services/integrations/models/sendgrid-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendgridConfigPayloadRequest = exports.sendgridConfigPayloadResponse = exports.sendgridConfigPayload = void 0;
const zod_1 = require("zod");
const sendgrid_config_payload_from_js_1 = require("./sendgrid-config-payload-from.js");
const sendgrid_config_payload_reply_to_js_1 = require("./sendgrid-config-payload-reply-to.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.sendgridConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        apiKey: zod_1.z.string(),
        from: sendgrid_config_payload_from_js_1.sendgridConfigPayloadFrom.optional(),
        replyTo: sendgrid_config_payload_reply_to_js_1.sendgridConfigPayloadReplyTo.optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.sendgridConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        api_key: zod_1.z.string(),
        from: sendgrid_config_payload_from_js_1.sendgridConfigPayloadFromResponse.optional(),
        reply_to: sendgrid_config_payload_reply_to_js_1.sendgridConfigPayloadReplyToResponse.optional(),
    })
        .transform((data) => ({
        apiKey: data['api_key'],
        from: data['from'],
        replyTo: data['reply_to'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.sendgridConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        apiKey: zod_1.z.string(),
        from: sendgrid_config_payload_from_js_1.sendgridConfigPayloadFromRequest.optional(),
        replyTo: sendgrid_config_payload_reply_to_js_1.sendgridConfigPayloadReplyToRequest.optional(),
    })
        .transform((data) => ({
        api_key: data['apiKey'],
        from: data['from'],
        reply_to: data['replyTo'],
    }));
});
