// magicbell-js/dist/commonjs/project-client/services/integrations/models/ses-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sesConfigRequest = exports.sesConfigResponse = exports.sesConfig = void 0;
const zod_1 = require("zod");
const ses_config_payload_js_1 = require("./ses-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.sesConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: ses_config_payload_js_1.sesConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.sesConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: ses_config_payload_js_1.sesConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.sesConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: ses_config_payload_js_1.sesConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
