// magicbell-js/dist/commonjs/project-client/services/integrations/models/ses-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sesConfigPayloadRequest = exports.sesConfigPayloadResponse = exports.sesConfigPayload = void 0;
const zod_1 = require("zod");
const ses_config_payload_from_js_1 = require("./ses-config-payload-from.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.sesConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        from: ses_config_payload_from_js_1.sesConfigPayloadFrom.optional(),
        keyId: zod_1.z.string().min(1),
        region: zod_1.z.string().min(1),
        secretKey: zod_1.z.string().min(1),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.sesConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        from: ses_config_payload_from_js_1.sesConfigPayloadFromResponse.optional(),
        key_id: zod_1.z.string().min(1),
        region: zod_1.z.string().min(1),
        secret_key: zod_1.z.string().min(1),
    })
        .transform((data) => ({
        from: data['from'],
        keyId: data['key_id'],
        region: data['region'],
        secretKey: data['secret_key'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.sesConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        from: ses_config_payload_from_js_1.sesConfigPayloadFromRequest.optional(),
        keyId: zod_1.z.string().min(1),
        region: zod_1.z.string().min(1),
        secretKey: zod_1.z.string().min(1),
    })
        .transform((data) => ({
        from: data['from'],
        key_id: data['keyId'],
        region: data['region'],
        secret_key: data['secretKey'],
    }));
});
