// magicbell-js/dist/commonjs/project-client/services/integrations/models/slack-bot-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackBotConfigRequest = exports.slackBotConfigResponse = exports.slackBotConfig = void 0;
const zod_1 = require("zod");
const slack_bot_config_payload_js_1 = require("./slack-bot-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackBotConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: slack_bot_config_payload_js_1.slackBotConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackBotConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: slack_bot_config_payload_js_1.slackBotConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackBotConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: slack_bot_config_payload_js_1.slackBotConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
