// magicbell-js/dist/commonjs/project-client/services/integrations/models/slack-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackConfigPayloadRequest = exports.slackConfigPayloadResponse = exports.slackConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z.string().regex(/^[0-9A-Z]+$/),
        clientId: zod_1.z.string().regex(/^[0-9]+\.[0-9]+$/),
        clientSecret: zod_1.z.string().min(32).max(32),
        signingSecret: zod_1.z.string().min(32).max(32),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z.string().regex(/^[0-9A-Z]+$/),
        client_id: zod_1.z.string().regex(/^[0-9]+\.[0-9]+$/),
        client_secret: zod_1.z.string().min(32).max(32),
        signing_secret: zod_1.z.string().min(32).max(32),
    })
        .transform((data) => ({
        appId: data['app_id'],
        clientId: data['client_id'],
        clientSecret: data['client_secret'],
        signingSecret: data['signing_secret'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z.string().regex(/^[0-9A-Z]+$/),
        clientId: zod_1.z.string().regex(/^[0-9]+\.[0-9]+$/),
        clientSecret: zod_1.z.string().min(32).max(32),
        signingSecret: zod_1.z.string().min(32).max(32),
    })
        .transform((data) => ({
        app_id: data['appId'],
        client_id: data['clientId'],
        client_secret: data['clientSecret'],
        signing_secret: data['signingSecret'],
    }));
});
