// magicbell-js/dist/commonjs/project-client/services/integrations/models/smtp-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.smtpConfigRequest = exports.smtpConfigResponse = exports.smtpConfig = void 0;
const zod_1 = require("zod");
const smtp_config_from_js_1 = require("./smtp-config-from.js");
const smtp_config_reply_to_js_1 = require("./smtp-config-reply-to.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.smtpConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        from: smtp_config_from_js_1.smtpConfigFrom,
        host: zod_1.z.string(),
        password: zod_1.z.string(),
        port: zod_1.z.number().gte(1).lte(65535),
        replyTo: smtp_config_reply_to_js_1.smtpConfigReplyTo.optional(),
        security: zod_1.z.string().optional(),
        username: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.smtpConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        from: smtp_config_from_js_1.smtpConfigFromResponse,
        host: zod_1.z.string(),
        password: zod_1.z.string(),
        port: zod_1.z.number().gte(1).lte(65535),
        reply_to: smtp_config_reply_to_js_1.smtpConfigReplyToResponse.optional(),
        security: zod_1.z.string().optional(),
        username: zod_1.z.string(),
    })
        .transform((data) => ({
        from: data['from'],
        host: data['host'],
        password: data['password'],
        port: data['port'],
        replyTo: data['reply_to'],
        security: data['security'],
        username: data['username'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.smtpConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        from: smtp_config_from_js_1.smtpConfigFromRequest,
        host: zod_1.z.string(),
        password: zod_1.z.string(),
        port: zod_1.z.number().gte(1).lte(65535),
        replyTo: smtp_config_reply_to_js_1.smtpConfigReplyToRequest.optional(),
        security: zod_1.z.string().optional(),
        username: zod_1.z.string(),
    })
        .transform((data) => ({
        from: data['from'],
        host: data['host'],
        password: data['password'],
        port: data['port'],
        reply_to: data['replyTo'],
        security: data['security'],
        username: data['username'],
    }));
});
