// magicbell-js/dist/commonjs/project-client/services/integrations/models/smtp-config-object.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.smtpConfigObjectRequest = exports.smtpConfigObjectResponse = exports.smtpConfigObject = void 0;
const zod_1 = require("zod");
const smtp_config_js_1 = require("./smtp-config.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.smtpConfigObject = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: smtp_config_js_1.smtpConfig,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.smtpConfigObjectResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: smtp_config_js_1.smtpConfigResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.smtpConfigObjectRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: smtp_config_js_1.smtpConfigRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
