// magicbell-js/dist/commonjs/project-client/services/integrations/models/stripe-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stripeConfigPayloadRequest = exports.stripeConfigPayloadResponse = exports.stripeConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.stripeConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        id: zod_1.z.string().optional(),
        webhookSigningSecret: zod_1.z.string().min(1).max(100),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.stripeConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        id: zod_1.z.string().optional(),
        webhook_signing_secret: zod_1.z.string().min(1).max(100),
    })
        .transform((data) => ({
        id: data['id'],
        webhookSigningSecret: data['webhook_signing_secret'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.stripeConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        id: zod_1.z.string().optional(),
        webhookSigningSecret: zod_1.z.string().min(1).max(100),
    })
        .transform((data) => ({
        id: data['id'],
        webhook_signing_secret: data['webhookSigningSecret'],
    }));
});
