// magicbell-js/dist/commonjs/project-client/services/integrations/models/stripe-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stripeConfigRequest = exports.stripeConfigResponse = exports.stripeConfig = void 0;
const zod_1 = require("zod");
const stripe_config_payload_js_1 = require("./stripe-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.stripeConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: stripe_config_payload_js_1.stripeConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.stripeConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: stripe_config_payload_js_1.stripeConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.stripeConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: stripe_config_payload_js_1.stripeConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
