// magicbell-js/dist/commonjs/project-client/services/integrations/models/twilio-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.twilioConfigRequest = exports.twilioConfigResponse = exports.twilioConfig = void 0;
const zod_1 = require("zod");
const twilio_config_payload_js_1 = require("./twilio-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.twilioConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: twilio_config_payload_js_1.twilioConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.twilioConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: twilio_config_payload_js_1.twilioConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.twilioConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: twilio_config_payload_js_1.twilioConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
