// magicbell-js/dist/commonjs/project-client/services/integrations/models/twilio-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.twilioConfigPayloadRequest = exports.twilioConfigPayloadResponse = exports.twilioConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.twilioConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        accountSid: zod_1.z.string().min(1).max(100),
        apiKey: zod_1.z.string().min(1).max(100),
        apiSecret: zod_1.z.string().min(1).max(100),
        from: zod_1.z
            .string()
            .min(1)
            .max(100)
            .regex(/^\+[0-9]{1,14}$/),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.twilioConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        account_sid: zod_1.z.string().min(1).max(100),
        api_key: zod_1.z.string().min(1).max(100),
        api_secret: zod_1.z.string().min(1).max(100),
        from: zod_1.z
            .string()
            .min(1)
            .max(100)
            .regex(/^\+[0-9]{1,14}$/),
    })
        .transform((data) => ({
        accountSid: data['account_sid'],
        apiKey: data['api_key'],
        apiSecret: data['api_secret'],
        from: data['from'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.twilioConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        accountSid: zod_1.z.string().min(1).max(100),
        apiKey: zod_1.z.string().min(1).max(100),
        apiSecret: zod_1.z.string().min(1).max(100),
        from: zod_1.z
            .string()
            .min(1)
            .max(100)
            .regex(/^\+[0-9]{1,14}$/),
    })
        .transform((data) => ({
        account_sid: data['accountSid'],
        api_key: data['apiKey'],
        api_secret: data['apiSecret'],
        from: data['from'],
    }));
});
