// magicbell-js/dist/commonjs/project-client/services/integrations/models/unread.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unreadRequest = exports.unreadResponse = exports.unread = void 0;
const zod_1 = require("zod");
const unread_hover_js_1 = require("./unread-hover.js");
const unread_state_js_1 = require("./unread-state.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.unread = zod_1.z.lazy(() => {
    return zod_1.z.object({
        backgroundColor: zod_1.z.string(),
        hover: unread_hover_js_1.unreadHover.optional(),
        state: unread_state_js_1.unreadState.optional(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.unreadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        hover: unread_hover_js_1.unreadHoverResponse.optional(),
        state: unread_state_js_1.unreadStateResponse.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        hover: data['hover'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.unreadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        hover: unread_hover_js_1.unreadHoverRequest.optional(),
        state: unread_state_js_1.unreadStateRequest.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        hover: data['hover'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
