// magicbell-js/dist/commonjs/project-client/services/integrations/models/webpush-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.webpushConfigRequest = exports.webpushConfigResponse = exports.webpushConfig = void 0;
const zod_1 = require("zod");
const webpush_config_payload_js_1 = require("./webpush-config-payload.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.webpushConfig = zod_1.z.lazy(() => {
    return zod_1.z.object({
        config: webpush_config_payload_js_1.webpushConfigPayload,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.webpushConfigResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: webpush_config_payload_js_1.webpushConfigPayloadResponse,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.webpushConfigRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        config: webpush_config_payload_js_1.webpushConfigPayloadRequest,
        id: zod_1.z.string(),
        name: zod_1.z.string(),
    })
        .transform((data) => ({
        config: data['config'],
        id: data['id'],
        name: data['name'],
    }));
});
