// magicbell-js/dist/commonjs/project-client/services/integrations/models/webpush-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.webpushConfigPayloadRequest = exports.webpushConfigPayloadResponse = exports.webpushConfigPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.webpushConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        privateKey: zod_1.z.string().min(8).max(128),
        publicKey: zod_1.z.string().min(8).max(128),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.webpushConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        private_key: zod_1.z.string().min(8).max(128),
        public_key: zod_1.z.string().min(8).max(128),
    })
        .transform((data) => ({
        privateKey: data['private_key'],
        publicKey: data['public_key'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.webpushConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        privateKey: zod_1.z.string().min(8).max(128),
        publicKey: zod_1.z.string().min(8).max(128),
    })
        .transform((data) => ({
        private_key: data['privateKey'],
        public_key: data['publicKey'],
    }));
});
