// magicbell-js/dist/esm/project-client/services/users/users-service.js
import { z } from 'zod';
import { Environment } from '../../http/environment.js';
import { RequestBuilder } from '../../http/transport/request-builder.js';
import { ContentType } from '../../http/types.js';
import { BaseService } from '../base-service.js';
import { userRequest, userResponse } from '../common/user.js';
import { userCollectionResponse } from './models/user-collection.js';
export class UsersService extends BaseService {
    /**
     * Lists all users in the project.
     * @param {number} [params.limit] -
     * @param {string} [params.startingAfter] -
     * @param {string} [params.endingBefore] -
     * @param {string} [params.query] -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<UserCollection>>} - OK
     */
    async listUsers(params, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('GET')
            .setPath('/users')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: userCollectionResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addQueryParam({
            key: 'limit',
            value: params === null || params === void 0 ? void 0 : params.limit,
        })
            .addQueryParam({
            key: 'starting_after',
            value: params === null || params === void 0 ? void 0 : params.startingAfter,
        })
            .addQueryParam({
            key: 'ending_before',
            value: params === null || params === void 0 ? void 0 : params.endingBefore,
        })
            .addQueryParam({
            key: 'query',
            value: params === null || params === void 0 ? void 0 : params.query,
        })
            .build();
        return this.client.call(request);
    }
    /**
     * Creates or updates a user with the provided details. The user will be associated with the project specified in the request context.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<User>>} - OK
     */
    async saveUser(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('PUT')
            .setPath('/users')
            .setRequestSchema(userRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: userResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Removes a user and all associated data from the project.
     * @param {string} userId -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<any>>} - No Content
     */
    async deleteUser(userId, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('DELETE')
            .setPath('/users/{user_id}')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: z.undefined(),
            contentType: ContentType.NoContent,
            status: 204,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addPathParam({
            key: 'user_id',
            value: userId,
        })
            .build();
        return this.client.call(request);
    }
}
