// magicbell-js/dist/commonjs/project-client/services/broadcasts/models/broadcast.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.broadcastRequest = exports.broadcastResponse = exports.broadcast = void 0;
const zod_1 = require("zod");
const user_js_1 = require("../../common/user.js");
const broadcast_status_js_1 = require("./broadcast-status.js");
const overrides_js_1 = require("./overrides.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.broadcast = zod_1.z.lazy(() => {
    return zod_1.z.object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        category: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        createdAt: zod_1.z.string().optional(),
        customAttributes: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string().optional(),
        overrides: overrides_js_1.overrides.optional().nullable(),
        recipients: zod_1.z.array(user_js_1.user).min(1).max(1000).nullable(),
        status: broadcast_status_js_1.broadcastStatus.optional(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.broadcastResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        action_url: zod_1.z.string().max(2048).optional().nullable(),
        category: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        created_at: zod_1.z.string().optional(),
        custom_attributes: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string().optional(),
        overrides: overrides_js_1.overridesResponse.optional().nullable(),
        recipients: zod_1.z.array(user_js_1.userResponse).min(1).max(1000).nullable(),
        status: broadcast_status_js_1.broadcastStatusResponse.optional(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
    })
        .transform((data) => ({
        actionUrl: data['action_url'],
        category: data['category'],
        content: data['content'],
        createdAt: data['created_at'],
        customAttributes: data['custom_attributes'],
        id: data['id'],
        overrides: data['overrides'],
        recipients: data['recipients'],
        status: data['status'],
        title: data['title'],
        topic: data['topic'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.broadcastRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        category: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
        content: zod_1.z.string().max(10485760).optional().nullable(),
        createdAt: zod_1.z.string().optional(),
        customAttributes: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string().optional(),
        overrides: overrides_js_1.overridesRequest.optional().nullable(),
        recipients: zod_1.z.array(user_js_1.userRequest).min(1).max(1000).nullable(),
        status: broadcast_status_js_1.broadcastStatusRequest.optional(),
        title: zod_1.z.string().min(1).max(255),
        topic: zod_1.z
            .string()
            .max(255)
            .regex(/^[A-Za-z0-9_\.\-\/:]+$/)
            .optional()
            .nullable(),
    })
        .transform((data) => ({
        action_url: data['actionUrl'],
        category: data['category'],
        content: data['content'],
        created_at: data['createdAt'],
        custom_attributes: data['customAttributes'],
        id: data['id'],
        overrides: data['overrides'],
        recipients: data['recipients'],
        status: data['status'],
        title: data['title'],
        topic: data['topic'],
    }));
});
