// magicbell-js/dist/commonjs/project-client/services/broadcasts/models/broadcast-status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.broadcastStatusRequest = exports.broadcastStatusResponse = exports.broadcastStatus = void 0;
const zod_1 = require("zod");
const errors_js_1 = require("./errors.js");
const summary_js_1 = require("./summary.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.broadcastStatus = zod_1.z.lazy(() => {
    return zod_1.z.object({
        errors: zod_1.z.array(errors_js_1.errors).nullable(),
        status: zod_1.z.string(),
        summary: summary_js_1.summary,
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.broadcastStatusResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        errors: zod_1.z.array(errors_js_1.errorsResponse).nullable(),
        status: zod_1.z.string(),
        summary: summary_js_1.summaryResponse,
    })
        .transform((data) => ({
        errors: data['errors'],
        status: data['status'],
        summary: data['summary'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.broadcastStatusRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        errors: zod_1.z.array(errors_js_1.errorsRequest).nullable(),
        status: zod_1.z.string(),
        summary: summary_js_1.summaryRequest,
    })
        .transform((data) => ({
        errors: data['errors'],
        status: data['status'],
        summary: data['summary'],
    }));
});
