// magicbell-js/dist/commonjs/project-client/services/broadcasts/models/in-app.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.inAppRequest = exports.inAppResponse = exports.inApp = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.inApp = zod_1.z.lazy(() => {
    return zod_1.z.object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        content: zod_1.z.string().max(1048576).optional(),
        title: zod_1.z.string().min(1).max(255).optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.inAppResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        action_url: zod_1.z.string().max(2048).optional().nullable(),
        content: zod_1.z.string().max(1048576).optional(),
        title: zod_1.z.string().min(1).max(255).optional(),
    })
        .transform((data) => ({
        actionUrl: data['action_url'],
        content: data['content'],
        title: data['title'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.inAppRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        actionUrl: zod_1.z.string().max(2048).optional().nullable(),
        content: zod_1.z.string().max(1048576).optional(),
        title: zod_1.z.string().min(1).max(255).optional(),
    })
        .transform((data) => ({
        action_url: data['actionUrl'],
        content: data['content'],
        title: data['title'],
    }));
});
