// magicbell-js/dist/commonjs/project-client/services/broadcasts/models/overrides-channels.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.overridesChannelsRequest = exports.overridesChannelsResponse = exports.overridesChannels = void 0;
const zod_1 = require("zod");
const email_js_1 = require("./email.js");
const in_app_js_1 = require("./in-app.js");
const mobile_push_js_1 = require("./mobile-push.js");
const sms_js_1 = require("./sms.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.overridesChannels = zod_1.z.lazy(() => {
    return zod_1.z.object({
        email: email_js_1.email.optional(),
        inApp: in_app_js_1.inApp.optional(),
        mobilePush: mobile_push_js_1.mobilePush.optional(),
        sms: sms_js_1.sms.optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.overridesChannelsResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        email: email_js_1.emailResponse.optional(),
        in_app: in_app_js_1.inAppResponse.optional(),
        mobile_push: mobile_push_js_1.mobilePushResponse.optional(),
        sms: sms_js_1.smsResponse.optional(),
    })
        .transform((data) => ({
        email: data['email'],
        inApp: data['in_app'],
        mobilePush: data['mobile_push'],
        sms: data['sms'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.overridesChannelsRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        email: email_js_1.emailRequest.optional(),
        inApp: in_app_js_1.inAppRequest.optional(),
        mobilePush: mobile_push_js_1.mobilePushRequest.optional(),
        sms: sms_js_1.smsRequest.optional(),
    })
        .transform((data) => ({
        email: data['email'],
        in_app: data['inApp'],
        mobile_push: data['mobilePush'],
        sms: data['sms'],
    }));
});
