// magicbell-js/dist/commonjs/project-client/services/broadcasts/models/providers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.providersRequest = exports.providersResponse = exports.providers = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.providers = zod_1.z.lazy(() => {
    return zod_1.z.object({
        apns: zod_1.z.any().optional(),
        expo: zod_1.z.any().optional(),
        fcm: zod_1.z.any().optional(),
        mailgun: zod_1.z.any().optional(),
        sendgrid: zod_1.z.any().optional(),
        ses: zod_1.z.any().optional(),
        slack: zod_1.z.any().optional(),
        teams: zod_1.z.any().optional(),
        twilio: zod_1.z.any().optional(),
        webPush: zod_1.z.any().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.providersResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        apns: zod_1.z.any().optional(),
        expo: zod_1.z.any().optional(),
        fcm: zod_1.z.any().optional(),
        mailgun: zod_1.z.any().optional(),
        sendgrid: zod_1.z.any().optional(),
        ses: zod_1.z.any().optional(),
        slack: zod_1.z.any().optional(),
        teams: zod_1.z.any().optional(),
        twilio: zod_1.z.any().optional(),
        web_push: zod_1.z.any().optional(),
    })
        .transform((data) => ({
        apns: data['apns'],
        expo: data['expo'],
        fcm: data['fcm'],
        mailgun: data['mailgun'],
        sendgrid: data['sendgrid'],
        ses: data['ses'],
        slack: data['slack'],
        teams: data['teams'],
        twilio: data['twilio'],
        webPush: data['web_push'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.providersRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        apns: zod_1.z.any().optional(),
        expo: zod_1.z.any().optional(),
        fcm: zod_1.z.any().optional(),
        mailgun: zod_1.z.any().optional(),
        sendgrid: zod_1.z.any().optional(),
        ses: zod_1.z.any().optional(),
        slack: zod_1.z.any().optional(),
        teams: zod_1.z.any().optional(),
        twilio: zod_1.z.any().optional(),
        webPush: zod_1.z.any().optional(),
    })
        .transform((data) => ({
        apns: data['apns'],
        expo: data['expo'],
        fcm: data['fcm'],
        mailgun: data['mailgun'],
        sendgrid: data['sendgrid'],
        ses: data['ses'],
        slack: data['slack'],
        teams: data['teams'],
        twilio: data['twilio'],
        web_push: data['webPush'],
    }));
});
