// magicbell-js/dist/commonjs/user-client/services/channels/models/apns-token-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.apnsTokenPayloadRequest = exports.apnsTokenPayloadResponse = exports.apnsTokenPayload = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.apnsTokenPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        deviceToken: zod_1.z.string().min(64),
        installationId: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsTokenPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        device_token: zod_1.z.string().min(64),
        installation_id: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        deviceToken: data['device_token'],
        installationId: data['installation_id'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsTokenPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        deviceToken: zod_1.z.string().min(64),
        installationId: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        device_token: data['deviceToken'],
        installation_id: data['installationId'],
    }));
});
