// magicbell-js/dist/commonjs/user-client/services/channels/models/discard-result.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.discardResultRequest = exports.discardResultResponse = exports.discardResult = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.discardResult = zod_1.z.lazy(() => {
    return zod_1.z.object({
        discardedAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.discardResultResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        discarded_at: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        discardedAt: data['discarded_at'],
        id: data['id'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.discardResultRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        discardedAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        discarded_at: data['discardedAt'],
        id: data['id'],
    }));
});
