// magicbell-js/dist/commonjs/user-client/services/channels/models/inbox-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.inboxTokenRequest = exports.inboxTokenResponse1 = exports.inboxToken = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.inboxToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        connectionId: zod_1.z.string().optional().nullable(),
        token: zod_1.z.string().min(64),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxTokenResponse1 = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        connection_id: zod_1.z.string().optional().nullable(),
        token: zod_1.z.string().min(64),
    })
        .transform((data) => ({
        connectionId: data['connection_id'],
        token: data['token'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        connectionId: zod_1.z.string().optional().nullable(),
        token: zod_1.z.string().min(64),
    })
        .transform((data) => ({
        connection_id: data['connectionId'],
        token: data['token'],
    }));
});
