// magicbell-js/dist/commonjs/user-client/services/channels/models/slack-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackTokenRequest = exports.slackTokenResponse = exports.slackToken = void 0;
const zod_1 = require("zod");
const slack_token_oauth_js_1 = require("./slack-token-oauth.js");
const slack_token_webhook_js_1 = require("./slack-token-webhook.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        oauth: slack_token_oauth_js_1.slackTokenOauth.optional(),
        updatedAt: zod_1.z.string().optional().nullable(),
        webhook: slack_token_webhook_js_1.slackTokenWebhook.optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string(),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        oauth: slack_token_oauth_js_1.slackTokenOauthResponse.optional(),
        updated_at: zod_1.z.string().optional().nullable(),
        webhook: slack_token_webhook_js_1.slackTokenWebhookResponse.optional(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        oauth: data['oauth'],
        updatedAt: data['updated_at'],
        webhook: data['webhook'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        oauth: slack_token_oauth_js_1.slackTokenOauthRequest.optional(),
        updatedAt: zod_1.z.string().optional().nullable(),
        webhook: slack_token_webhook_js_1.slackTokenWebhookRequest.optional(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        oauth: data['oauth'],
        updated_at: data['updatedAt'],
        webhook: data['webhook'],
    }));
});
