// magicbell-js/dist/commonjs/user-client/services/channels/models/slack-token-payload-oauth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackTokenPayloadOauthRequest = exports.slackTokenPayloadOauthResponse = exports.slackTokenPayloadOauth = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackTokenPayloadOauth = zod_1.z.lazy(() => {
    return zod_1.z.object({
        channelId: zod_1.z.string(),
        installationId: zod_1.z.string(),
        scope: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackTokenPayloadOauthResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        channel_id: zod_1.z.string(),
        installation_id: zod_1.z.string(),
        scope: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        channelId: data['channel_id'],
        installationId: data['installation_id'],
        scope: data['scope'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackTokenPayloadOauthRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        channelId: zod_1.z.string(),
        installationId: zod_1.z.string(),
        scope: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        channel_id: data['channelId'],
        installation_id: data['installationId'],
        scope: data['scope'],
    }));
});
