// magicbell-js/dist/commonjs/user-client/services/channels/models/teams-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.teamsTokenRequest = exports.teamsTokenResponse = exports.teamsToken = void 0;
const zod_1 = require("zod");
const teams_token_webhook_js_1 = require("./teams-token-webhook.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.teamsToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updatedAt: zod_1.z.string().optional().nullable(),
        webhook: teams_token_webhook_js_1.teamsTokenWebhook.optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.teamsTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string(),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updated_at: zod_1.z.string().optional().nullable(),
        webhook: teams_token_webhook_js_1.teamsTokenWebhookResponse.optional(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        updatedAt: data['updated_at'],
        webhook: data['webhook'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.teamsTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updatedAt: zod_1.z.string().optional().nullable(),
        webhook: teams_token_webhook_js_1.teamsTokenWebhookRequest.optional(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        updated_at: data['updatedAt'],
        webhook: data['webhook'],
    }));
});
