// magicbell-js/dist/commonjs/user-client/services/channels/models/web-push-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.webPushTokenRequest = exports.webPushTokenResponse = exports.webPushToken = void 0;
const zod_1 = require("zod");
const web_push_token_keys_js_1 = require("./web-push-token-keys.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.webPushToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        endpoint: zod_1.z.string(),
        id: zod_1.z.string(),
        keys: web_push_token_keys_js_1.webPushTokenKeys,
        updatedAt: zod_1.z.string().optional().nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.webPushTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string(),
        discarded_at: zod_1.z.string().optional().nullable(),
        endpoint: zod_1.z.string(),
        id: zod_1.z.string(),
        keys: web_push_token_keys_js_1.webPushTokenKeysResponse,
        updated_at: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        discardedAt: data['discarded_at'],
        endpoint: data['endpoint'],
        id: data['id'],
        keys: data['keys'],
        updatedAt: data['updated_at'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.webPushTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        endpoint: zod_1.z.string(),
        id: zod_1.z.string(),
        keys: web_push_token_keys_js_1.webPushTokenKeysRequest,
        updatedAt: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        discarded_at: data['discardedAt'],
        endpoint: data['endpoint'],
        id: data['id'],
        keys: data['keys'],
        updated_at: data['updatedAt'],
    }));
});
