// magicbell-js/dist/esm/project-client/services/events/events-service.js
import { z } from 'zod';
import { Environment } from '../../http/environment.js';
import { RequestBuilder } from '../../http/transport/request-builder.js';
import { ContentType } from '../../http/types.js';
import { BaseService } from '../base-service.js';
import { eventResponse } from './models/event.js';
import { eventCollectionResponse } from './models/event-collection.js';
export class EventsService extends BaseService {
    /**
     * Retrieves a paginated list of events for the project.
     * @param {number} [params.limit] -
     * @param {string} [params.startingAfter] -
     * @param {string} [params.endingBefore] -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<EventCollection>>} - OK
     */
    async listEvents(params, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('GET')
            .setPath('/events')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: eventCollectionResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addQueryParam({
            key: 'limit',
            value: params === null || params === void 0 ? void 0 : params.limit,
        })
            .addQueryParam({
            key: 'starting_after',
            value: params === null || params === void 0 ? void 0 : params.startingAfter,
        })
            .addQueryParam({
            key: 'ending_before',
            value: params === null || params === void 0 ? void 0 : params.endingBefore,
        })
            .build();
        return this.client.call(request);
    }
    /**
     * Fetches a project event by its ID.
     * @param {string} eventId -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<Event>>} - OK
     */
    async fetchEvent(eventId, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('GET')
            .setPath('/events/{event_id}')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: eventResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addPathParam({
            key: 'event_id',
            value: eventId,
        })
            .build();
        return this.client.call(request);
    }
}
