// magicbell-js/dist/commonjs/project-client/services/workflows/models/get-run-response-status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRunResponseStatusRequest = exports.getRunResponseStatusResponse = exports.getRunResponseStatus = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.getRunResponseStatus = zod_1.z.lazy(() => {
    return zod_1.z.object({
        completedAt: zod_1.z.string().optional(),
        error: zod_1.z.string().optional(),
        nextStep: zod_1.z.number().optional(),
        startedAt: zod_1.z.string().optional(),
        state: zod_1.z.number().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.getRunResponseStatusResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        completed_at: zod_1.z.string().optional(),
        error: zod_1.z.string().optional(),
        next_step: zod_1.z.number().optional(),
        started_at: zod_1.z.string().optional(),
        state: zod_1.z.number().optional(),
    })
        .transform((data) => ({
        completedAt: data['completed_at'],
        error: data['error'],
        nextStep: data['next_step'],
        startedAt: data['started_at'],
        state: data['state'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.getRunResponseStatusRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        completedAt: zod_1.z.string().optional(),
        error: zod_1.z.string().optional(),
        nextStep: zod_1.z.number().optional(),
        startedAt: zod_1.z.string().optional(),
        state: zod_1.z.number().optional(),
    })
        .transform((data) => ({
        completed_at: data['completedAt'],
        error: data['error'],
        next_step: data['nextStep'],
        started_at: data['startedAt'],
        state: data['state'],
    }));
});
