// magicbell-js/dist/commonjs/project-client/services/workflows/models/get-run-response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRunResponseRequest = exports.getRunResponseResponse = exports.getRunResponse = void 0;
const zod_1 = require("zod");
const get_run_response_status_js_1 = require("./get-run-response-status.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.getRunResponse = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: get_run_response_status_js_1.getRunResponseStatus.optional(),
        workflowKey: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.getRunResponseResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: get_run_response_status_js_1.getRunResponseStatusResponse.optional(),
        workflow_key: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        id: data['id'],
        status: data['status'],
        workflowKey: data['workflow_key'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.getRunResponseRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: get_run_response_status_js_1.getRunResponseStatusRequest.optional(),
        workflowKey: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        id: data['id'],
        status: data['status'],
        workflow_key: data['workflowKey'],
    }));
});
