// magicbell-js/dist/commonjs/project-client/services/workflows/models/workflow-run.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.workflowRunRequest = exports.workflowRunResponse = exports.workflowRun = void 0;
const zod_1 = require("zod");
const workflow_run_status_js_1 = require("./workflow-run-status.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.workflowRun = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: workflow_run_status_js_1.workflowRunStatus.optional(),
        workflowKey: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.workflowRunResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: workflow_run_status_js_1.workflowRunStatusResponse.optional(),
        workflow_key: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        id: data['id'],
        status: data['status'],
        workflowKey: data['workflow_key'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.workflowRunRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string().optional(),
        id: zod_1.z.string().optional(),
        status: workflow_run_status_js_1.workflowRunStatusRequest.optional(),
        workflowKey: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        id: data['id'],
        status: data['status'],
        workflow_key: data['workflowKey'],
    }));
});
