// magicbell-js/dist/esm/user-client/index.js
import { ChannelsService } from './services/channels/index.js';
import { IntegrationsService } from './services/integrations/index.js';
import { NotificationsService } from './services/notifications/index.js';
export { Environment } from './http/environment.js';
export * from './http/index.js';
export * from './services/channels/index.js';
export * from './services/common/index.js';
export * from './services/integrations/index.js';
export * from './services/notifications/index.js';
export class Client {
    constructor(config) {
        Object.defineProperty(this, "config", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: config
        });
        Object.defineProperty(this, "channels", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "integrations", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "notifications", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.channels = new ChannelsService(this.config);
        this.integrations = new IntegrationsService(this.config);
        this.notifications = new NotificationsService(this.config);
    }
    set baseUrl(baseUrl) {
        this.channels.baseUrl = baseUrl;
        this.integrations.baseUrl = baseUrl;
        this.notifications.baseUrl = baseUrl;
    }
    set environment(environment) {
        this.channels.baseUrl = environment;
        this.integrations.baseUrl = environment;
        this.notifications.baseUrl = environment;
    }
    set timeoutMs(timeoutMs) {
        this.channels.timeoutMs = timeoutMs;
        this.integrations.timeoutMs = timeoutMs;
        this.notifications.timeoutMs = timeoutMs;
    }
    set token(token) {
        this.channels.token = token;
        this.integrations.token = token;
        this.notifications.token = token;
    }
}
// c029837e0e474b76bc487506e8799df5e3335891efe4fb02bda7a1441840310c
