// magicbell-js/dist/commonjs/project-client/services/broadcasts/broadcasts-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BroadcastsService = void 0;
const zod_1 = require("zod");
const environment_js_1 = require("../../http/environment.js");
const request_builder_js_1 = require("../../http/transport/request-builder.js");
const types_js_1 = require("../../http/types.js");
const base_service_js_1 = require("../base-service.js");
const broadcast_js_1 = require("./models/broadcast.js");
const broadcast_collection_js_1 = require("./models/broadcast-collection.js");
class BroadcastsService extends base_service_js_1.BaseService {
    /**
     * Retrieves a paginated list of broadcasts for the project. Returns basic information about each broadcast including its creation time and status.
     * @param {number} [params.limit] -
     * @param {string} [params.startingAfter] -
     * @param {string} [params.endingBefore] -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<BroadcastCollection>>} - OK
     */
    async listBroadcasts(params, requestConfig) {
        const request = new request_builder_js_1.RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || environment_js_1.Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('GET')
            .setPath('/broadcasts')
            .setRequestSchema(zod_1.z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(types_js_1.ContentType.Json)
            .addResponse({
            schema: broadcast_collection_js_1.broadcastCollectionResponse,
            contentType: types_js_1.ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addQueryParam({
            key: 'limit',
            value: params === null || params === void 0 ? void 0 : params.limit,
        })
            .addQueryParam({
            key: 'starting_after',
            value: params === null || params === void 0 ? void 0 : params.startingAfter,
        })
            .addQueryParam({
            key: 'ending_before',
            value: params === null || params === void 0 ? void 0 : params.endingBefore,
        })
            .build();
        return this.client.call(request);
    }
    /**
     * Creates a new broadcast. When a broadcast is created, it generates individual notifications for relevant users within the project.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<Broadcast>>} - Created
     */
    async createBroadcast(body, requestConfig) {
        const request = new request_builder_js_1.RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || environment_js_1.Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/broadcasts')
            .setRequestSchema(broadcast_js_1.broadcastRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(types_js_1.ContentType.Json)
            .addResponse({
            schema: broadcast_js_1.broadcastResponse,
            contentType: types_js_1.ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Retrieves detailed information about a specific broadcast by its ID. Includes the broadcast's configuration and current status.
     * @param {string} broadcastId -
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<Broadcast>>} - OK
     */
    async fetchBroadcast(broadcastId, requestConfig) {
        const request = new request_builder_js_1.RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || environment_js_1.Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('GET')
            .setPath('/broadcasts/{broadcast_id}')
            .setRequestSchema(zod_1.z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(types_js_1.ContentType.Json)
            .addResponse({
            schema: broadcast_js_1.broadcastResponse,
            contentType: types_js_1.ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addPathParam({
            key: 'broadcast_id',
            value: broadcastId,
        })
            .build();
        return this.client.call(request);
    }
}
exports.BroadcastsService = BroadcastsService;
