// magicbell-js/dist/esm/user-client/services/integrations/integrations-service.js
import { z } from 'zod';
import { Environment } from '../../http/environment.js';
import { RequestBuilder } from '../../http/transport/request-builder.js';
import { ContentType } from '../../http/types.js';
import { BaseService } from '../base-service.js';
import { webPushTokenPayloadRequest, webPushTokenPayloadResponse, } from '../common/web-push-token-payload.js';
import { inboxConfigPayloadRequest, inboxConfigPayloadResponse, } from './models/inbox-config-payload.js';
import { slackFinishInstallResponseRequest, } from './models/slack-finish-install-response.js';
import { slackInstallationRequest, slackInstallationResponse } from './models/slack-installation.js';
import { slackStartInstallRequest } from './models/slack-start-install.js';
import { slackStartInstallResponseContentResponse, } from './models/slack-start-install-response-content.js';
import { webPushStartInstallationResponseResponse, } from './models/web-push-start-installation-response.js';
export class IntegrationsService extends BaseService {
    /**
     * Creates a new installation of a Inbox integration for a user. This endpoint is used when an integration needs to be set up with user-specific credentials or configuration.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<InboxConfigPayload>>} - OK
     */
    async saveInboxInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('PUT')
            .setPath('/integrations/inbox/installations')
            .setRequestSchema(inboxConfigPayloadRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: inboxConfigPayloadResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Initiates the installation flow for an Inbox integration. This is the first step in a multi-step installation process where user authorization or external service configuration may be required.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<InboxConfigPayload>>} - Created
     */
    async startInboxInstallation(requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/inbox/installations/start')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: inboxConfigPayloadResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .build();
        return this.client.call(request);
    }
    /**
     * Creates a new installation of a MagicBell SlackBot integration for a user. This endpoint is used when an integration needs to be set up with user-specific credentials or configuration.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackInstallation>>} - OK
     */
    async saveMagicbellSlackbotInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('PUT')
            .setPath('/integrations/magicbell_slackbot/installations')
            .setRequestSchema(slackInstallationRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackInstallationResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Completes the installation flow for the MagicBell SlackBot integration. This endpoint is typically called after the user has completed any required authorization steps with MagicBell SlackBot.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackInstallation>>} - Created
     */
    async finishMagicbellSlackbotInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/magicbell_slackbot/installations/finish')
            .setRequestSchema(slackFinishInstallResponseRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackInstallationResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Initiates the installation flow for a MagicBell SlackBot integration. This is the first step in a multi-step installation process where user authorization or external service configuration may be required.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackStartInstallResponseContent>>} - Created
     */
    async startMagicbellSlackbotInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/magicbell_slackbot/installations/start')
            .setRequestSchema(slackStartInstallRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackStartInstallResponseContentResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Creates a new installation of a Slack integration for a user. This endpoint is used when an integration needs to be set up with user-specific credentials or configuration.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackInstallation>>} - OK
     */
    async saveSlackInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('PUT')
            .setPath('/integrations/slack/installations')
            .setRequestSchema(slackInstallationRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackInstallationResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Completes the installation flow for the Slack integration. This endpoint is typically called after the user has completed any required authorization steps with Slack.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackInstallation>>} - Created
     */
    async finishSlackInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/slack/installations/finish')
            .setRequestSchema(slackFinishInstallResponseRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackInstallationResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Initiates the installation flow for a Slack integration. This is the first step in a multi-step installation process where user authorization or external service configuration may be required.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<SlackStartInstallResponseContent>>} - Created
     */
    async startSlackInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/slack/installations/start')
            .setRequestSchema(slackStartInstallRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: slackStartInstallResponseContentResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Creates a new installation of a Web Push integration for a user. This endpoint is used when an integration needs to be set up with user-specific credentials or configuration.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<WebPushTokenPayload>>} - OK
     */
    async saveWebPushInstallation(body, requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('PUT')
            .setPath('/integrations/web_push/installations')
            .setRequestSchema(webPushTokenPayloadRequest)
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: webPushTokenPayloadResponse,
            contentType: ContentType.Json,
            status: 200,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .addHeaderParam({ key: 'Content-Type', value: 'application/json' })
            .addBody(body)
            .build();
        return this.client.call(request);
    }
    /**
     * Initiates the installation flow for a Web Push integration. This is the first step in a multi-step installation process where user authorization or external service configuration may be required.
     * @param {RequestConfig} [requestConfig] - The request configuration for retry and validation.
     * @returns {Promise<HttpResponse<WebPushStartInstallationResponse>>} - Created
     */
    async startWebPushInstallation(requestConfig) {
        const request = new RequestBuilder()
            .setBaseUrl((requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.baseUrl) || this.config.baseUrl || this.config.environment || Environment.DEFAULT)
            .setConfig(this.config)
            .setMethod('POST')
            .setPath('/integrations/web_push/installations/start')
            .setRequestSchema(z.any())
            .addAccessTokenAuth(this.config.token, 'Bearer')
            .setRequestContentType(ContentType.Json)
            .addResponse({
            schema: webPushStartInstallationResponseResponse,
            contentType: ContentType.Json,
            status: 201,
        })
            .setRetryAttempts(this.config, requestConfig)
            .setRetryDelayMs(this.config, requestConfig)
            .setResponseValidation(this.config, requestConfig)
            .build();
        return this.client.call(request);
    }
}
