// magicbell-js/dist/commonjs/user-client/services/integrations/models/authed-user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.authedUserRequest = exports.authedUserResponse = exports.authedUser = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.authedUser = zod_1.z.lazy(() => {
    return zod_1.z.object({
        accessToken: zod_1.z.string().optional(),
        expiresIn: zod_1.z.number().optional(),
        id: zod_1.z.string(),
        refreshToken: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        tokenType: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.authedUserResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        access_token: zod_1.z.string().optional(),
        expires_in: zod_1.z.number().optional(),
        id: zod_1.z.string(),
        refresh_token: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        token_type: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        accessToken: data['access_token'],
        expiresIn: data['expires_in'],
        id: data['id'],
        refreshToken: data['refresh_token'],
        scope: data['scope'],
        tokenType: data['token_type'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.authedUserRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        accessToken: zod_1.z.string().optional(),
        expiresIn: zod_1.z.number().optional(),
        id: zod_1.z.string(),
        refreshToken: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        tokenType: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        access_token: data['accessToken'],
        expires_in: data['expiresIn'],
        id: data['id'],
        refresh_token: data['refreshToken'],
        scope: data['scope'],
        token_type: data['tokenType'],
    }));
});
