// magicbell-js/dist/commonjs/user-client/services/integrations/models/dialog.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.dialogRequest = exports.dialogResponse = exports.dialog = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.dialog = zod_1.z.lazy(() => {
    return zod_1.z.object({
        accentColor: zod_1.z.string(),
        backgroundColor: zod_1.z.string(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.dialogResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        accentColor: zod_1.z.string(),
        backgroundColor: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        accentColor: data['accentColor'],
        backgroundColor: data['backgroundColor'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.dialogRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        accentColor: zod_1.z.string(),
        backgroundColor: zod_1.z.string(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        accentColor: data['accentColor'],
        backgroundColor: data['backgroundColor'],
        textColor: data['textColor'],
    }));
});
