// magicbell-js/dist/commonjs/user-client/services/integrations/models/inbox-config-payload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.inboxConfigPayloadRequest = exports.inboxConfigPayloadResponse = exports.inboxConfigPayload = void 0;
const zod_1 = require("zod");
const images_js_1 = require("./images.js");
const theme_js_1 = require("./theme.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.inboxConfigPayload = zod_1.z.lazy(() => {
    return zod_1.z.object({
        images: images_js_1.images.nullable(),
        locale: zod_1.z.string().min(2).nullable(),
        theme: theme_js_1.theme.nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxConfigPayloadResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        images: images_js_1.imagesResponse.nullable(),
        locale: zod_1.z.string().min(2).nullable(),
        theme: theme_js_1.themeResponse.nullable(),
    })
        .transform((data) => ({
        images: data['images'],
        locale: data['locale'],
        theme: data['theme'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxConfigPayloadRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        images: images_js_1.imagesRequest.nullable(),
        locale: zod_1.z.string().min(2).nullable(),
        theme: theme_js_1.themeRequest.nullable(),
    })
        .transform((data) => ({
        images: data['images'],
        locale: data['locale'],
        theme: data['theme'],
    }));
});
