// magicbell-js/dist/commonjs/user-client/services/integrations/models/incoming-webhook.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.incomingWebhookRequest = exports.incomingWebhookResponse = exports.incomingWebhook = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.incomingWebhook = zod_1.z.lazy(() => {
    return zod_1.z.object({
        channel: zod_1.z.string(),
        configurationUrl: zod_1.z.string(),
        url: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.incomingWebhookResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        channel: zod_1.z.string(),
        configuration_url: zod_1.z.string(),
        url: zod_1.z.string(),
    })
        .transform((data) => ({
        channel: data['channel'],
        configurationUrl: data['configuration_url'],
        url: data['url'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.incomingWebhookRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        channel: zod_1.z.string(),
        configurationUrl: zod_1.z.string(),
        url: zod_1.z.string(),
    })
        .transform((data) => ({
        channel: data['channel'],
        configuration_url: data['configurationUrl'],
        url: data['url'],
    }));
});
