// magicbell-js/dist/commonjs/user-client/services/integrations/models/slack-start-install-response-content.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackStartInstallResponseContentRequest = exports.slackStartInstallResponseContentResponse = exports.slackStartInstallResponseContent = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackStartInstallResponseContent = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z.string().optional(),
        authUrl: zod_1.z.string().optional(),
        scopes: zod_1.z.array(zod_1.z.string()).optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackStartInstallResponseContentResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z.string().optional(),
        auth_url: zod_1.z.string().optional(),
        scopes: zod_1.z.array(zod_1.z.string()).optional(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        authUrl: data['auth_url'],
        scopes: data['scopes'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackStartInstallResponseContentRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z.string().optional(),
        authUrl: zod_1.z.string().optional(),
        scopes: zod_1.z.array(zod_1.z.string()).optional(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        auth_url: data['authUrl'],
        scopes: data['scopes'],
    }));
});
