// magicbell-js/dist/commonjs/user-client/services/integrations/models/slack-start-install.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackStartInstallRequest = exports.slackStartInstallResponse = exports.slackStartInstall = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackStartInstall = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z.string(),
        authUrl: zod_1.z.string().optional(),
        extraScopes: zod_1.z.array(zod_1.z.string()).optional(),
        redirectUrl: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackStartInstallResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z.string(),
        auth_url: zod_1.z.string().optional(),
        extra_scopes: zod_1.z.array(zod_1.z.string()).optional(),
        redirect_url: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        authUrl: data['auth_url'],
        extraScopes: data['extra_scopes'],
        redirectUrl: data['redirect_url'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackStartInstallRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z.string(),
        authUrl: zod_1.z.string().optional(),
        extraScopes: zod_1.z.array(zod_1.z.string()).optional(),
        redirectUrl: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        auth_url: data['authUrl'],
        extra_scopes: data['extraScopes'],
        redirect_url: data['redirectUrl'],
    }));
});
