// magicbell-js/dist/commonjs/user-client/services/integrations/models/slack-installation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackInstallationRequest = exports.slackInstallationResponse = exports.slackInstallation = void 0;
const zod_1 = require("zod");
const authed_user_js_1 = require("./authed-user.js");
const enterprise_js_1 = require("./enterprise.js");
const incoming_webhook_js_1 = require("./incoming-webhook.js");
const team_js_1 = require("./team.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackInstallation = zod_1.z.lazy(() => {
    return zod_1.z.object({
        accessToken: zod_1.z.string(),
        appId: zod_1.z.string(),
        authedUser: authed_user_js_1.authedUser,
        botUserId: zod_1.z.string().optional(),
        enterprise: enterprise_js_1.enterprise.optional(),
        expiresIn: zod_1.z.number().optional(),
        id: zod_1.z
            .string()
            .regex(/^[A-Z0-9]+-.*$/)
            .optional(),
        incomingWebhook: incoming_webhook_js_1.incomingWebhook.optional(),
        isEnterpriseInstall: zod_1.z.boolean().optional(),
        refreshToken: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        team: team_js_1.team,
        tokenType: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackInstallationResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        access_token: zod_1.z.string(),
        app_id: zod_1.z.string(),
        authed_user: authed_user_js_1.authedUserResponse,
        bot_user_id: zod_1.z.string().optional(),
        enterprise: enterprise_js_1.enterpriseResponse.optional(),
        expires_in: zod_1.z.number().optional(),
        id: zod_1.z
            .string()
            .regex(/^[A-Z0-9]+-.*$/)
            .optional(),
        incoming_webhook: incoming_webhook_js_1.incomingWebhookResponse.optional(),
        is_enterprise_install: zod_1.z.boolean().optional(),
        refresh_token: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        team: team_js_1.teamResponse,
        token_type: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        accessToken: data['access_token'],
        appId: data['app_id'],
        authedUser: data['authed_user'],
        botUserId: data['bot_user_id'],
        enterprise: data['enterprise'],
        expiresIn: data['expires_in'],
        id: data['id'],
        incomingWebhook: data['incoming_webhook'],
        isEnterpriseInstall: data['is_enterprise_install'],
        refreshToken: data['refresh_token'],
        scope: data['scope'],
        team: data['team'],
        tokenType: data['token_type'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackInstallationRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        accessToken: zod_1.z.string(),
        appId: zod_1.z.string(),
        authedUser: authed_user_js_1.authedUserRequest,
        botUserId: zod_1.z.string().optional(),
        enterprise: enterprise_js_1.enterpriseRequest.optional(),
        expiresIn: zod_1.z.number().optional(),
        id: zod_1.z
            .string()
            .regex(/^[A-Z0-9]+-.*$/)
            .optional(),
        incomingWebhook: incoming_webhook_js_1.incomingWebhookRequest.optional(),
        isEnterpriseInstall: zod_1.z.boolean().optional(),
        refreshToken: zod_1.z.string().optional(),
        scope: zod_1.z.string().optional(),
        team: team_js_1.teamRequest,
        tokenType: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        access_token: data['accessToken'],
        app_id: data['appId'],
        authed_user: data['authedUser'],
        bot_user_id: data['botUserId'],
        enterprise: data['enterprise'],
        expires_in: data['expiresIn'],
        id: data['id'],
        incoming_webhook: data['incomingWebhook'],
        is_enterprise_install: data['isEnterpriseInstall'],
        refresh_token: data['refreshToken'],
        scope: data['scope'],
        team: data['team'],
        token_type: data['tokenType'],
    }));
});
