// magicbell-js/dist/commonjs/user-client/services/integrations/models/slack-finish-install-response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.slackFinishInstallResponseRequest = exports.slackFinishInstallResponseResponse = exports.slackFinishInstallResponse = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.slackFinishInstallResponse = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z.string(),
        code: zod_1.z.string(),
        redirectUrl: zod_1.z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackFinishInstallResponseResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z.string(),
        code: zod_1.z.string(),
        redirect_url: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        code: data['code'],
        redirectUrl: data['redirect_url'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.slackFinishInstallResponseRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z.string(),
        code: zod_1.z.string(),
        redirectUrl: zod_1.z.string().optional(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        code: data['code'],
        redirect_url: data['redirectUrl'],
    }));
});
