// magicbell-js/dist/commonjs/user-client/services/integrations/models/theme.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.themeRequest = exports.themeResponse = exports.theme = void 0;
const zod_1 = require("zod");
const banner_js_1 = require("./banner.js");
const dialog_js_1 = require("./dialog.js");
const footer_js_1 = require("./footer.js");
const header_js_1 = require("./header.js");
const icon_js_1 = require("./icon.js");
const theme_notification_js_1 = require("./theme-notification.js");
const unseen_badge_js_1 = require("./unseen-badge.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.theme = zod_1.z.lazy(() => {
    return zod_1.z.object({
        banner: banner_js_1.banner.optional(),
        dialog: dialog_js_1.dialog.optional(),
        footer: footer_js_1.footer.optional(),
        header: header_js_1.header.optional(),
        icon: icon_js_1.icon.optional(),
        notification: theme_notification_js_1.themeNotification.optional(),
        unseenBadge: unseen_badge_js_1.unseenBadge.optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.themeResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        banner: banner_js_1.bannerResponse.optional(),
        dialog: dialog_js_1.dialogResponse.optional(),
        footer: footer_js_1.footerResponse.optional(),
        header: header_js_1.headerResponse.optional(),
        icon: icon_js_1.iconResponse.optional(),
        notification: theme_notification_js_1.themeNotificationResponse.optional(),
        unseenBadge: unseen_badge_js_1.unseenBadgeResponse.optional(),
    })
        .transform((data) => ({
        banner: data['banner'],
        dialog: data['dialog'],
        footer: data['footer'],
        header: data['header'],
        icon: data['icon'],
        notification: data['notification'],
        unseenBadge: data['unseenBadge'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.themeRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        banner: banner_js_1.bannerRequest.optional(),
        dialog: dialog_js_1.dialogRequest.optional(),
        footer: footer_js_1.footerRequest.optional(),
        header: header_js_1.headerRequest.optional(),
        icon: icon_js_1.iconRequest.optional(),
        notification: theme_notification_js_1.themeNotificationRequest.optional(),
        unseenBadge: unseen_badge_js_1.unseenBadgeRequest.optional(),
    })
        .transform((data) => ({
        banner: data['banner'],
        dialog: data['dialog'],
        footer: data['footer'],
        header: data['header'],
        icon: data['icon'],
        notification: data['notification'],
        unseenBadge: data['unseenBadge'],
    }));
});
