// magicbell-js/dist/commonjs/user-client/services/integrations/models/unseen.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.unseenRequest = exports.unseenResponse = exports.unseen = void 0;
const zod_1 = require("zod");
const unseen_hover_js_1 = require("./unseen-hover.js");
const unseen_state_js_1 = require("./unseen-state.js");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.unseen = zod_1.z.lazy(() => {
    return zod_1.z.object({
        backgroundColor: zod_1.z.string(),
        hover: unseen_hover_js_1.unseenHover.optional(),
        state: unseen_state_js_1.unseenState.optional(),
        textColor: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.unseenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        hover: unseen_hover_js_1.unseenHoverResponse.optional(),
        state: unseen_state_js_1.unseenStateResponse.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        hover: data['hover'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.unseenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        backgroundColor: zod_1.z.string(),
        hover: unseen_hover_js_1.unseenHoverRequest.optional(),
        state: unseen_state_js_1.unseenStateRequest.optional(),
        textColor: zod_1.z.string(),
    })
        .transform((data) => ({
        backgroundColor: data['backgroundColor'],
        hover: data['hover'],
        state: data['state'],
        textColor: data['textColor'],
    }));
});
