// magicbell-js/dist/commonjs/user-client/services/integrations/models/web-push-start-installation-response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.webPushStartInstallationResponseRequest = exports.webPushStartInstallationResponseResponse = exports.webPushStartInstallationResponse = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.webPushStartInstallationResponse = zod_1.z.lazy(() => {
    return zod_1.z.object({
        authToken: zod_1.z.string().min(8),
        publicKey: zod_1.z.string().min(8).max(128),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.webPushStartInstallationResponseResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        auth_token: zod_1.z.string().min(8),
        public_key: zod_1.z.string().min(8).max(128),
    })
        .transform((data) => ({
        authToken: data['auth_token'],
        publicKey: data['public_key'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.webPushStartInstallationResponseRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        authToken: zod_1.z.string().min(8),
        publicKey: zod_1.z.string().min(8).max(128),
    })
        .transform((data) => ({
        auth_token: data['authToken'],
        public_key: data['publicKey'],
    }));
});
