// magicbell-js/dist/commonjs/project-client/services/common/user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userRequest = exports.userResponse = exports.user = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.user = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string().optional().nullable(),
        customAttributes: zod_1.z.any().optional().nullable(),
        email: zod_1.z.string().optional().nullable(),
        externalId: zod_1.z.string().optional().nullable(),
        firstName: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string().optional(),
        lastName: zod_1.z.string().optional().nullable(),
        lastNotifiedAt: zod_1.z.string().optional().nullable(),
        lastSeenAt: zod_1.z.string().optional().nullable(),
        updatedAt: zod_1.z.string().optional().nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.userResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string().optional().nullable(),
        custom_attributes: zod_1.z.any().optional().nullable(),
        email: zod_1.z.string().optional().nullable(),
        external_id: zod_1.z.string().optional().nullable(),
        first_name: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string().optional(),
        last_name: zod_1.z.string().optional().nullable(),
        last_notified_at: zod_1.z.string().optional().nullable(),
        last_seen_at: zod_1.z.string().optional().nullable(),
        updated_at: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        customAttributes: data['custom_attributes'],
        email: data['email'],
        externalId: data['external_id'],
        firstName: data['first_name'],
        id: data['id'],
        lastName: data['last_name'],
        lastNotifiedAt: data['last_notified_at'],
        lastSeenAt: data['last_seen_at'],
        updatedAt: data['updated_at'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.userRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string().optional().nullable(),
        customAttributes: zod_1.z.any().optional().nullable(),
        email: zod_1.z.string().optional().nullable(),
        externalId: zod_1.z.string().optional().nullable(),
        firstName: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string().optional(),
        lastName: zod_1.z.string().optional().nullable(),
        lastNotifiedAt: zod_1.z.string().optional().nullable(),
        lastSeenAt: zod_1.z.string().optional().nullable(),
        updatedAt: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        custom_attributes: data['customAttributes'],
        email: data['email'],
        external_id: data['externalId'],
        first_name: data['firstName'],
        id: data['id'],
        last_name: data['lastName'],
        last_notified_at: data['lastNotifiedAt'],
        last_seen_at: data['lastSeenAt'],
        updated_at: data['updatedAt'],
    }));
});
