// magicbell-js/dist/commonjs/project-client/services/channels/models/apns-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.apnsTokenRequest = exports.apnsTokenResponse = exports.apnsToken = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.apnsToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        appId: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        createdAt: zod_1.z.string(),
        deviceToken: zod_1.z.string().min(64),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        installationId: zod_1.z.string().optional(),
        updatedAt: zod_1.z.string().optional().nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        app_id: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        created_at: zod_1.z.string(),
        device_token: zod_1.z.string().min(64),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        installation_id: zod_1.z.string().optional(),
        updated_at: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        createdAt: data['created_at'],
        deviceToken: data['device_token'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        installationId: data['installation_id'],
        updatedAt: data['updated_at'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.apnsTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        appId: zod_1.z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        createdAt: zod_1.z.string(),
        deviceToken: zod_1.z.string().min(64),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        installationId: zod_1.z.string().optional(),
        updatedAt: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        created_at: data['createdAt'],
        device_token: data['deviceToken'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        installation_id: data['installationId'],
        updated_at: data['updatedAt'],
    }));
});
