// magicbell-js/dist/commonjs/project-client/services/channels/models/expo-token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.expoTokenRequest = exports.expoTokenResponse = exports.expoToken = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.expoToken = zod_1.z.lazy(() => {
    return zod_1.z.object({
        createdAt: zod_1.z.string(),
        deviceToken: zod_1.z.string().min(1),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updatedAt: zod_1.z.string().optional().nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.expoTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        created_at: zod_1.z.string(),
        device_token: zod_1.z.string().min(1),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updated_at: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        createdAt: data['created_at'],
        deviceToken: data['device_token'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        updatedAt: data['updated_at'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.expoTokenRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        createdAt: zod_1.z.string(),
        deviceToken: zod_1.z.string().min(1),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        updatedAt: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        created_at: data['createdAt'],
        device_token: data['deviceToken'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        updated_at: data['updatedAt'],
    }));
});
