// magicbell-js/dist/commonjs/project-client/services/channels/models/inbox-token-response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.inboxTokenResponseRequest = exports.inboxTokenResponseResponse = exports.inboxTokenResponse = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.inboxTokenResponse = zod_1.z.lazy(() => {
    return zod_1.z.object({
        connectionId: zod_1.z.string().optional().nullable(),
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        token: zod_1.z.string().min(64),
        updatedAt: zod_1.z.string().optional().nullable(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxTokenResponseResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        connection_id: zod_1.z.string().optional().nullable(),
        created_at: zod_1.z.string(),
        discarded_at: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        token: zod_1.z.string().min(64),
        updated_at: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        connectionId: data['connection_id'],
        createdAt: data['created_at'],
        discardedAt: data['discarded_at'],
        id: data['id'],
        token: data['token'],
        updatedAt: data['updated_at'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.inboxTokenResponseRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        connectionId: zod_1.z.string().optional().nullable(),
        createdAt: zod_1.z.string(),
        discardedAt: zod_1.z.string().optional().nullable(),
        id: zod_1.z.string(),
        token: zod_1.z.string().min(64),
        updatedAt: zod_1.z.string().optional().nullable(),
    })
        .transform((data) => ({
        connection_id: data['connectionId'],
        created_at: data['createdAt'],
        discarded_at: data['discardedAt'],
        id: data['id'],
        token: data['token'],
        updated_at: data['updatedAt'],
    }));
});
