// magicbell-js/dist/commonjs/project-client/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = exports.Environment = void 0;
const tslib_1 = require("tslib");
const index_js_1 = require("./services/broadcasts/index.js");
const index_js_2 = require("./services/channels/index.js");
const index_js_3 = require("./services/events/index.js");
const index_js_4 = require("./services/integrations/index.js");
const index_js_5 = require("./services/users/index.js");
const index_js_6 = require("./services/workflows/index.js");
var environment_js_1 = require("./http/environment.js");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return environment_js_1.Environment; } });
tslib_1.__exportStar(require("./http/index.js"), exports);
tslib_1.__exportStar(require("./services/broadcasts/index.js"), exports);
tslib_1.__exportStar(require("./services/channels/index.js"), exports);
tslib_1.__exportStar(require("./services/common/index.js"), exports);
tslib_1.__exportStar(require("./services/events/index.js"), exports);
tslib_1.__exportStar(require("./services/integrations/index.js"), exports);
tslib_1.__exportStar(require("./services/users/index.js"), exports);
tslib_1.__exportStar(require("./services/workflows/index.js"), exports);
class Client {
    constructor(config) {
        Object.defineProperty(this, "config", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: config
        });
        Object.defineProperty(this, "broadcasts", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "channels", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "events", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "integrations", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "users", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "workflows", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.broadcasts = new index_js_1.BroadcastsService(this.config);
        this.channels = new index_js_2.ChannelsService(this.config);
        this.events = new index_js_3.EventsService(this.config);
        this.integrations = new index_js_4.IntegrationsService(this.config);
        this.users = new index_js_5.UsersService(this.config);
        this.workflows = new index_js_6.WorkflowsService(this.config);
    }
    set baseUrl(baseUrl) {
        this.broadcasts.baseUrl = baseUrl;
        this.channels.baseUrl = baseUrl;
        this.events.baseUrl = baseUrl;
        this.integrations.baseUrl = baseUrl;
        this.users.baseUrl = baseUrl;
        this.workflows.baseUrl = baseUrl;
    }
    set environment(environment) {
        this.broadcasts.baseUrl = environment;
        this.channels.baseUrl = environment;
        this.events.baseUrl = environment;
        this.integrations.baseUrl = environment;
        this.users.baseUrl = environment;
        this.workflows.baseUrl = environment;
    }
    set timeoutMs(timeoutMs) {
        this.broadcasts.timeoutMs = timeoutMs;
        this.channels.timeoutMs = timeoutMs;
        this.events.timeoutMs = timeoutMs;
        this.integrations.timeoutMs = timeoutMs;
        this.users.timeoutMs = timeoutMs;
        this.workflows.timeoutMs = timeoutMs;
    }
    set token(token) {
        this.broadcasts.token = token;
        this.channels.token = token;
        this.events.token = token;
        this.integrations.token = token;
        this.users.token = token;
        this.workflows.token = token;
    }
}
exports.Client = Client;
// c029837e0e474b76bc487506e8799df5e3335891efe4fb02bda7a1441840310c
