// magicbell-js/dist/esm/project-client/http/transport/request-builder.js
import z from 'zod';
import { Environment } from '../environment.js';
import { SerializationStyle } from '../serialization/base-serializer.js';
import { ContentType } from '../types.js';
import { Request } from './request.js';
export class RequestBuilder {
    constructor() {
        Object.defineProperty(this, "params", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.params = {
            baseUrl: Environment.DEFAULT,
            method: 'GET',
            path: '',
            config: {},
            responses: [],
            errors: [],
            requestSchema: z.any(),
            requestContentType: ContentType.Json,
            retry: {
                attempts: 3,
                delayMs: 150,
            },
            validation: {
                responseValidation: true,
            },
            pathParams: new Map(),
            queryParams: new Map(),
            headers: new Map(),
        };
    }
    setRetryAttempts(sdkConfig, requestConfig) {
        var _a, _b;
        if (((_a = requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.retry) === null || _a === void 0 ? void 0 : _a.attempts) !== undefined) {
            this.params.retry.attempts = requestConfig.retry.attempts;
        }
        else if (((_b = sdkConfig === null || sdkConfig === void 0 ? void 0 : sdkConfig.retry) === null || _b === void 0 ? void 0 : _b.attempts) !== undefined) {
            this.params.retry.attempts = sdkConfig.retry.attempts;
        }
        return this;
    }
    setRetryDelayMs(sdkConfig, requestConfig) {
        var _a, _b;
        if (((_a = requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.retry) === null || _a === void 0 ? void 0 : _a.delayMs) !== undefined) {
            this.params.retry.delayMs = requestConfig.retry.delayMs;
        }
        else if (((_b = sdkConfig === null || sdkConfig === void 0 ? void 0 : sdkConfig.retry) === null || _b === void 0 ? void 0 : _b.delayMs) !== undefined) {
            this.params.retry.delayMs = sdkConfig.retry.delayMs;
        }
        return this;
    }
    setResponseValidation(sdkConfig, requestConfig) {
        var _a, _b;
        if (((_a = requestConfig === null || requestConfig === void 0 ? void 0 : requestConfig.validation) === null || _a === void 0 ? void 0 : _a.responseValidation) !== undefined) {
            this.params.validation.responseValidation = requestConfig.validation.responseValidation;
        }
        else if (((_b = sdkConfig === null || sdkConfig === void 0 ? void 0 : sdkConfig.validation) === null || _b === void 0 ? void 0 : _b.responseValidation) !== undefined) {
            this.params.validation.responseValidation = sdkConfig.validation.responseValidation;
        }
        return this;
    }
    setBaseUrl(baseUrl) {
        if (baseUrl) {
            this.params.baseUrl = baseUrl;
        }
        return this;
    }
    setMethod(method) {
        this.params.method = method;
        return this;
    }
    setPath(path) {
        this.params.path = path;
        return this;
    }
    setConfig(config) {
        this.params.config = config;
        return this;
    }
    setRequestContentType(contentType) {
        this.params.requestContentType = contentType;
        return this;
    }
    setRequestSchema(requestSchema) {
        this.params.requestSchema = requestSchema;
        return this;
    }
    setFilename(filename) {
        if (filename !== undefined) {
            this.params.filename = filename;
        }
        return this;
    }
    setFilenames(filenames) {
        if (filenames !== undefined) {
            this.params.filenames = filenames;
        }
        return this;
    }
    setPagination(pagination) {
        this.params.pagination = pagination;
        return this;
    }
    setCursorPagination(pagination) {
        this.params.pagination = pagination;
        return this;
    }
    addAccessTokenAuth(accessToken, prefix) {
        if (accessToken === undefined) {
            return this;
        }
        this.params.headers.set('Authorization', {
            key: 'Authorization',
            value: `${prefix !== null && prefix !== void 0 ? prefix : 'Bearer'} ${accessToken}`,
            explode: false,
            style: SerializationStyle.SIMPLE,
            encode: true,
            isLimit: false,
            isOffset: false,
            isCursor: false,
        });
        return this;
    }
    addBasicAuth(username, password) {
        if (username === undefined || password === undefined) {
            return this;
        }
        this.params.headers.set('Authorization', {
            key: 'Authorization',
            value: `Basic ${this.toBase64(`${username}:${password}`)}`,
            explode: false,
            style: SerializationStyle.SIMPLE,
            encode: true,
            isLimit: false,
            isOffset: false,
            isCursor: false,
        });
        return this;
    }
    addApiKeyAuth(apiKey, keyName) {
        if (apiKey === undefined) {
            return this;
        }
        this.params.headers.set(keyName !== null && keyName !== void 0 ? keyName : 'X-API-KEY', {
            key: keyName !== null && keyName !== void 0 ? keyName : 'X-API-KEY',
            value: apiKey,
            explode: false,
            style: SerializationStyle.SIMPLE,
            encode: true,
            isLimit: false,
            isOffset: false,
            isCursor: false,
        });
        return this;
    }
    addResponse(response) {
        this.params.responses.push(response);
        return this;
    }
    addError(error) {
        this.params.errors.push(error);
        return this;
    }
    addBody(body) {
        if (body !== undefined) {
            this.params.body = body;
        }
        return this;
    }
    addPathParam(param) {
        var _a, _b, _c;
        if (param.value === undefined || param.key === undefined) {
            return this;
        }
        this.params.pathParams.set(param.key, {
            key: param.key,
            value: param.value,
            explode: (_a = param.explode) !== null && _a !== void 0 ? _a : true,
            style: (_b = param.style) !== null && _b !== void 0 ? _b : SerializationStyle.SIMPLE,
            encode: (_c = param.encode) !== null && _c !== void 0 ? _c : true,
            isLimit: !!param.isLimit,
            isOffset: !!param.isOffset,
            isCursor: !!param.isCursor,
        });
        return this;
    }
    addQueryParam(param) {
        var _a, _b, _c;
        if (param.key === undefined) {
            return this;
        }
        this.params.queryParams.set(param.key, {
            key: param.key,
            value: param.value,
            explode: (_a = param.explode) !== null && _a !== void 0 ? _a : true,
            style: (_b = param.style) !== null && _b !== void 0 ? _b : SerializationStyle.FORM,
            encode: (_c = param.encode) !== null && _c !== void 0 ? _c : true,
            isLimit: !!param.isLimit,
            isOffset: !!param.isOffset,
            isCursor: !!param.isCursor,
        });
        return this;
    }
    addHeaderParam(param) {
        var _a, _b, _c;
        if (param.value === undefined || param.key === undefined) {
            return this;
        }
        this.params.headers.set(param.key, {
            key: param.key,
            value: param.value,
            explode: (_a = param.explode) !== null && _a !== void 0 ? _a : true,
            style: (_b = param.style) !== null && _b !== void 0 ? _b : SerializationStyle.SIMPLE,
            encode: (_c = param.encode) !== null && _c !== void 0 ? _c : false,
            isLimit: !!param.isLimit,
            isOffset: !!param.isOffset,
            isCursor: !!param.isCursor,
        });
        return this;
    }
    build() {
        return new Request(this.params);
    }
    toBase64(str) {
        if (typeof window === 'undefined') {
            return Buffer.from(str, 'utf-8').toString('base64');
        }
        else {
            return btoa(unescape(encodeURIComponent(str)));
        }
    }
}
