// magicbell-js/dist/esm/project-client/http/transport/request.js
import { SerializationStyle } from '../serialization/base-serializer.js';
import { HeaderSerializer } from '../serialization/header-serializer.js';
import { PathSerializer } from '../serialization/path-serializer.js';
import { QuerySerializer } from '../serialization/query-serializer.js';
import { isRequestCursorPagination, } from './types.js';
export class Request {
    constructor(params) {
        Object.defineProperty(this, "baseUrl", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: ''
        });
        Object.defineProperty(this, "headers", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new Map()
        });
        Object.defineProperty(this, "queryParams", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new Map()
        });
        Object.defineProperty(this, "pathParams", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: new Map()
        });
        Object.defineProperty(this, "body", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "method", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "path", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "config", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "responses", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "errors", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "requestSchema", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "requestContentType", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "validation", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {}
        });
        Object.defineProperty(this, "retry", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: {}
        });
        Object.defineProperty(this, "pagination", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "filename", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "filenames", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "pathPattern", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.baseUrl = params.baseUrl;
        this.method = params.method;
        this.pathPattern = params.path;
        this.body = params.body;
        this.path = this.constructPath();
        this.config = params.config;
        this.pathParams = params.pathParams;
        this.headers = params.headers;
        this.queryParams = params.queryParams;
        this.responses = params.responses;
        this.errors = params.errors;
        this.requestSchema = params.requestSchema;
        this.requestContentType = params.requestContentType;
        this.retry = params.retry;
        this.validation = params.validation;
        this.pagination = params.pagination;
        this.filename = params.filename;
        this.filenames = params.filenames;
    }
    addHeaderParam(key, param) {
        if (param.value === undefined) {
            return;
        }
        if (param.explode === undefined) {
            param.explode = false;
        }
        if (param.style === undefined) {
            param.style = SerializationStyle.SIMPLE;
        }
        if (param.encode === undefined) {
            param.encode = false;
        }
        this.headers.set(key, param);
    }
    addQueryParam(key, param) {
        if (param.value === undefined) {
            return;
        }
        if (param.explode === undefined) {
            param.explode = true;
        }
        if (param.style === undefined) {
            param.style = SerializationStyle.FORM;
        }
        if (param.encode === undefined) {
            param.encode = true;
        }
        this.queryParams.set(key, param);
    }
    addPathParam(key, param) {
        if (param.value === undefined) {
            return;
        }
        if (param.explode === undefined) {
            param.explode = false;
        }
        if (param.style === undefined) {
            param.style = SerializationStyle.SIMPLE;
        }
        if (param.encode === undefined) {
            param.encode = true;
        }
        this.pathParams.set(key, param);
    }
    addBody(body) {
        if (body === undefined) {
            return;
        }
        this.body = body;
    }
    updateFromHookRequest(hookRequest) {
        this.baseUrl = hookRequest.baseUrl;
        this.method = hookRequest.method;
        this.path = hookRequest.path;
        this.body = hookRequest.body;
    }
    constructFullUrl() {
        const queryString = new QuerySerializer().serialize(this.queryParams);
        const path = this.constructPath();
        const baseUrl = this.baseUrl;
        return `${baseUrl}${path}${queryString}`;
    }
    copy(overrides) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r;
        const createRequestParams = {
            baseUrl: (_a = overrides === null || overrides === void 0 ? void 0 : overrides.baseUrl) !== null && _a !== void 0 ? _a : this.baseUrl,
            errors: (_b = overrides === null || overrides === void 0 ? void 0 : overrides.errors) !== null && _b !== void 0 ? _b : this.errors,
            method: (_c = overrides === null || overrides === void 0 ? void 0 : overrides.method) !== null && _c !== void 0 ? _c : this.method,
            path: (_d = overrides === null || overrides === void 0 ? void 0 : overrides.path) !== null && _d !== void 0 ? _d : this.path,
            body: (_e = overrides === null || overrides === void 0 ? void 0 : overrides.body) !== null && _e !== void 0 ? _e : this.body,
            config: (_f = overrides === null || overrides === void 0 ? void 0 : overrides.config) !== null && _f !== void 0 ? _f : this.config,
            pathParams: (_g = overrides === null || overrides === void 0 ? void 0 : overrides.pathParams) !== null && _g !== void 0 ? _g : this.pathParams,
            queryParams: (_h = overrides === null || overrides === void 0 ? void 0 : overrides.queryParams) !== null && _h !== void 0 ? _h : this.queryParams,
            headers: (_j = overrides === null || overrides === void 0 ? void 0 : overrides.headers) !== null && _j !== void 0 ? _j : this.headers,
            responses: (_k = overrides === null || overrides === void 0 ? void 0 : overrides.responses) !== null && _k !== void 0 ? _k : this.responses,
            requestSchema: (_l = overrides === null || overrides === void 0 ? void 0 : overrides.requestSchema) !== null && _l !== void 0 ? _l : this.requestSchema,
            requestContentType: (_m = overrides === null || overrides === void 0 ? void 0 : overrides.requestContentType) !== null && _m !== void 0 ? _m : this.requestContentType,
            retry: (_o = overrides === null || overrides === void 0 ? void 0 : overrides.retry) !== null && _o !== void 0 ? _o : this.retry,
            validation: (_p = overrides === null || overrides === void 0 ? void 0 : overrides.validation) !== null && _p !== void 0 ? _p : this.validation,
            filename: (_q = overrides === null || overrides === void 0 ? void 0 : overrides.filename) !== null && _q !== void 0 ? _q : this.filename,
            filenames: (_r = overrides === null || overrides === void 0 ? void 0 : overrides.filenames) !== null && _r !== void 0 ? _r : this.filenames,
        };
        return new Request({
            ...createRequestParams,
            ...overrides,
        });
    }
    getHeaders() {
        if (!this.headers || !this.headers.size) {
            return undefined;
        }
        return new HeaderSerializer().serialize(this.headers);
    }
    nextPage(cursor) {
        if (!this.pagination) {
            return;
        }
        // Check if this is cursor pagination using type guard
        if (isRequestCursorPagination(this.pagination)) {
            const cursorParam = this.getCursorParam();
            if (cursorParam && cursor !== undefined) {
                cursorParam.value = cursor;
            }
            return;
        }
        // Handle limit-offset pagination
        const offsetParam = this.getOffsetParam();
        if (offsetParam) {
            if (this.pagination.pageSize === undefined) {
                throw new Error('pageSize is required for limit-offset pagination');
            }
            offsetParam.value = Number(offsetParam.value) + this.pagination.pageSize;
        }
    }
    constructPath() {
        return new PathSerializer().serialize(this.pathPattern, this.pathParams);
    }
    getOffsetParam() {
        const offsetParam = this.getAllParams().find((param) => param.isOffset);
        return offsetParam;
    }
    getCursorParam() {
        const cursorParam = this.getAllParams().find((param) => param.isCursor);
        return cursorParam;
    }
    getAllParams() {
        const allParams = [];
        this.headers.forEach((val, _) => {
            allParams.push(val);
        });
        this.queryParams.forEach((val, _) => {
            allParams.push(val);
        });
        this.pathParams.forEach((val, _) => {
            allParams.push(val);
        });
        return allParams;
    }
}
