// magicbell-js/dist/commonjs/project-client/services/events/models/event.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.eventRequest = exports.eventResponse = exports.event = void 0;
const zod_1 = require("zod");
/**
 * The shape of the model inside the application code - what the users use
 */
exports.event = zod_1.z.lazy(() => {
    return zod_1.z.object({
        code: zod_1.z.number().optional(),
        context: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string(),
        level: zod_1.z.string().optional(),
        log: zod_1.z.string().optional().nullable(),
        timestamp: zod_1.z.string(),
        type: zod_1.z.string(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
exports.eventResponse = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        code: zod_1.z.number().optional(),
        context: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string(),
        level: zod_1.z.string().optional(),
        log: zod_1.z.string().optional().nullable(),
        timestamp: zod_1.z.string(),
        type: zod_1.z.string(),
    })
        .transform((data) => ({
        code: data['code'],
        context: data['context'],
        id: data['id'],
        level: data['level'],
        log: data['log'],
        timestamp: data['timestamp'],
        type: data['type'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
exports.eventRequest = zod_1.z.lazy(() => {
    return zod_1.z
        .object({
        code: zod_1.z.number().optional(),
        context: zod_1.z.any().optional().nullable(),
        id: zod_1.z.string(),
        level: zod_1.z.string().optional(),
        log: zod_1.z.string().optional().nullable(),
        timestamp: zod_1.z.string(),
        type: zod_1.z.string(),
    })
        .transform((data) => ({
        code: data['code'],
        context: data['context'],
        id: data['id'],
        level: data['level'],
        log: data['log'],
        timestamp: data['timestamp'],
        type: data['type'],
    }));
});
