// magicbell-js/dist/esm/user-client/services/channels/models/apns-token-payload.js
import { z } from 'zod';
/**
 * The shape of the model inside the application code - what the users use
 */
export const apnsTokenPayload = z.lazy(() => {
    return z.object({
        appId: z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        deviceToken: z.string().min(64),
        installationId: z.string().optional(),
    });
});
/**
 * The shape of the model mapping from the api schema into the application shape.
 * Is equal to application shape if all property names match the api schema
 */
export const apnsTokenPayloadResponse = z.lazy(() => {
    return z
        .object({
        app_id: z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        device_token: z.string().min(64),
        installation_id: z.string().optional(),
    })
        .transform((data) => ({
        appId: data['app_id'],
        deviceToken: data['device_token'],
        installationId: data['installation_id'],
    }));
});
/**
 * The shape of the model mapping from the application shape into the api schema.
 * Is equal to application shape if all property names match the api schema
 */
export const apnsTokenPayloadRequest = z.lazy(() => {
    return z
        .object({
        appId: z
            .string()
            .regex(/^[a-zA-Z0-9]+(.[a-zA-Z0-9]+)*$/)
            .optional(),
        deviceToken: z.string().min(64),
        installationId: z.string().optional(),
    })
        .transform((data) => ({
        app_id: data['appId'],
        device_token: data['deviceToken'],
        installation_id: data['installationId'],
    }));
});
